chrome/content/zotero/zoteroPane.js

```diff
@@ -136,8 +139,182 @@ var ZoteroPane = new function()
 		
 		// continue loading pane
 		_loadPane();
+		setUpToolbar();
 	};
-	
+
+	function setUpToolbar() {
+		/* constants, state, and helper functions */
+		const toolbar = this.document.getElementById("zotero-toolbar");
+		const lookupButton = document.getElementById("zotero-tb-lookup"); // needs special treatment
+		let zone = null;
+		let current = null;
+
+		class TbFocusZone {
+			constructor(node, before, after) {
+				this.node = node;
+				this.before = before;
+				this.after = after;
+			}
+			enter(targetButton) {
+				zone = this;
+				current = targetButton;
+			}
+			exit() {
+				zone = null;
+				current = null;
+			}
+			getButtons() {
+				return Array.from(this.node.getElementsByClassName("zotero-tb-button"));
+			}
+			get firstButton() {
+				return this.getButtons()[0];
+			}
+			contains(target) {
+				return this.node.contains(target);
+			}
+		}
+		
+		function moveFocus(n = 0) {
+			const children = zone.getButtons();
+			let currentIndex = children.indexOf(current);
+			if (currentIndex === -1) {
+				throw new Error("Cannot move focus from an element not part of the current toolbar zone.");
+			}
+			if (currentIndex + n < 0 || currentIndex + n >= children.length) return;
+			children[currentIndex + n].focus();
+		}
+		
+		const isTbButton = (node) => node && node.classList && node.classList.contains("zotero-tb-button");
+		
+		/* set up toolbar */
+
+		const zones = [
+			new TbFocusZone(document.getElementById("zotero-collections-toolbar") ),
+			new TbFocusZone(document.getElementById("zotero-items-toolbar"),
+								() => zones[0].firstButton,
+								() => document.getElementById("zotero-tb-search-menu-button")),
+			new TbFocusZone(document.getElementById("zotero-item-toolbar"),
+								() => document.getElementById("collection-tree"),
+								() => document.getElementById("zotero-tb-search"))
```

Comment:
Nit: Three parameters on separate lines, single tab for each:

```javascript
new TbFocusZone(
    document.getElementById("zotero-item-toolbar"),
    () => document.getElementById("collection-tree"),
    () => document.getElementById("zotero-tb-search")
)
```

Reply:
Also, `zotero-tb-search` and `collection-tree` are backwards, so if you press ArrowDown from the search button, you end up in the search box.